cookiecutter/generate.py

```diff
@@ -81,19 +103,24 @@ def generate_context(
     """
     context = OrderedDict([])
 
-    try:
-        with open(context_file, encoding='utf-8') as file_handle:
-            obj = json.load(file_handle, object_pairs_hook=OrderedDict)
-    except ValueError as e:
-        # JSON decoding error.  Let's throw a new exception that is more
-        # friendly for the developer or user.
-        full_fpath = os.path.abspath(context_file)
-        json_exc_message = str(e)
-        our_exc_message = (
-            'JSON decoding error while loading "{0}".  Decoding'
-            ' error details: "{1}"'.format(full_fpath, json_exc_message)
-        )
-        raise ContextDecodingException(our_exc_message)
+    if context_file.endswith('.json'):
+        try:
+            with open(context_file, encoding='utf-8') as file_handle:
+                obj = json.load(file_handle, object_pairs_hook=OrderedDict)
+        except ValueError as e:
+            # JSON decoding error.  Let's throw a new exception that is more
+            # friendly for the developer or user.
+            full_fpath = os.path.abspath(context_file)
+            json_exc_message = str(e)
+            our_exc_message = (
+                'JSON decoding error while loading "{0}".  Decoding'
+                ' error details: "{1}"'.format(full_fpath, json_exc_message)
+            )
+            raise ContextDecodingException(our_exc_message)
+    else:
+        # FIXME handle potential errors
+        with open(context_file) as file_handle:
+            obj = ordered_load(file_handle, yaml.SafeLoader)
```

Comment:
I do not think we should try to dress the exceptions, it would not make sense in JSON context. Some errors are specific to the loader and should be kept unmodified.